File: Is Binary Number Multiple of 3[30-June-2023]/solution.py
#  >> PYTHON CODE <<

class Solution:
	def isDivisible(self, s):
	    sum=0
	    for i in range(len(s)):
	        if(s[i]=='1'):
	            if(i%2==0):
	                sum+=1
	            else:
	                sum+=2
        
        return 1 if sum%3==0 else 0

File: Is Binary Number Multiple of 3[30-June-2023]/solution.cpp
//   >> C++ CODE <<

int isDivisible(string s) {
    int sum = 0;
    for (int i = 0; i < s.length(); i++) {
        if (s[i] == '1') {
            if (i % 2 == 0) {
                sum += 1;
            } else {
                sum += 2;
            }
        }
    }
    
    return sum % 3 == 0 ? 1 : 0;
}